### ISSUE TITLE
linkerd proxy continually logging: trust_dns_proto::xfer error notifying wait, possible future leak: Err(ProtoError { inner: requestor canceled })

### ISSUE THREAD

**Issue by nicksc423** [linkerd/linkerd2#3842]
### What is the issue?
I have a k8s cluster in AKS with an nginx-ingress controller and linkerd stable-2.6.0.  The linkerd proxy in my main-nginx-ingress controller pod is continually logging the following WARN message:
`WARN [1012267.254916s] trust_dns_proto::xfer error notifying wait, possible future leak: Err(ProtoError { inner: 
requestor canceled })  `
### How can it be reproduced?
I do not know
#### `linkerd check` output

```
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-config
--------------
√ control plane Namespace exists
√ control plane ClusterRoles exist
√ control plane ClusterRoleBindings exist
√ control plane ServiceAccounts exist
√ control plane CustomResourceDefinitions exist
√ control plane MutatingWebhookConfigurations exist
√ control plane ValidatingWebhookConfigurations exist
√ control plane PodSecurityPolicies exist

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ heartbeat ServiceAccount exist
√ control plane replica sets are ready
√ no unschedulable pods
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-api
-----------
√ control plane pods are ready
√ control plane self-check
√ [kubernetes] control plane can talk to Kubernetes
√ [prometheus] control plane can talk to Prometheus
√ no invalid service profiles

linkerd-version
---------------
√ can determine the latest version
‼ cli is up-to-date
    is running version 2.6.0 but the latest stable version is 2.6.1
    see https://linkerd.io/checks/#l5d-version-cli for hints

control-plane-version
---------------------
‼ control plane is up-to-date
    is running version 2.6.0 but the latest stable version is 2.6.1
    see https://linkerd.io/checks/#l5d-version-control for hints
√ control plane and cli versions match

Status check results are √
```

### Environment

- Kubernetes Version: v1.14.8
- Cluster Environment: AKS
- Host OS: ubuntu 16.04
- Linkerd version: stable-2.6.0

### Additional context

This is a dev cluster that has barely anything running on it.  This makes it difficult to tell if there are any resource leaks.  I mention this because the only other context I can find this error is [issue 2012](https://github.com/linkerd/linkerd2/issues/2012) which mentions a memory leak.    I do see memory constantly rising for the proxy on my cluster, but since there's barely anything running on the cluster it's just a 2MB rise which could easily just be normal.  

![Screen Shot 2019-12-17 at 2 24 21 PM](https://user-images.githubusercontent.com/17151642/71044045-6908b780-20e5-11ea-9bd5-166da4eacc56.png)

**Comment by grampelberg** [comment]
@nicksc423 mind trying out the latest edge? We don't have replication steps for this and are hoping you can see if it has gone away.

**Comment by grampelberg** [comment]
This should be fixed now with `stable-2.7.1`. Please reopen if you're continuing to see this issue!
